core/server/data/migration/populate.js

```diff
@@ -30,20 +32,32 @@ populate = function populate(options) {
             context: {
                 internal: true
             }
-        },
-        tableSequence = Promise.mapSeries(schemaTables, function createTable(table) {
-            logger.info('Creating table: ' + table);
-            return commands.createTable(table);
-        });
+        };
 
     logger.info('Creating tables...');
 
-    if (tablesOnly) {
-        return tableSequence;
-    }
+    return new Promise(function populateDatabase(resolve, reject) {
+        db.knex.transaction(function populateDatabaseInTransaction(transaction) {
+            return Promise.mapSeries(schemaTables, function createTable(table) {
+                logger.info('Creating table: ' + table);
+                return commands.createTable(table, transaction);
+            }).then(function populateFixtures() {
+                if (tablesOnly) {
+                    return;
+                }
 
-    return tableSequence.then(function () {
-        return fixtures.populate(logger, modelOptions);
+                return fixtures.populate(logger, _.merge({}, {transacting: transaction}, modelOptions));
+            }).then(function commitTransaction() {
+                transaction.commit();
+                resolve();
+            }).catch(function rollbackTransaction(err) {
+                logger.warn('rolling back because of: ' + err.stack);
+
+                transaction.rollback();
+            });
+        }).catch(function populateDatabaseError() {
+            reject();
```

Comment:
yeah this was weird. when i started using knex transactions there was no other solution then this one. i think i can see why now, because i didn't return the Promise, see https://github.com/TryGhost/Ghost/blob/master/core/server/data/migration/update.js#L61

But here we return the Promise. yeah i think the handling here i wrong now. will change now 🎉